### ISSUE TITLE
index out of bounds in decoder

### ISSUE THREAD

**Issue by batyrmastyr** [xmonader/nim-bencode#1]
The `source[idx..source.len]` in following lines in bencodeparser.nim
```
158 let pair = this.decode_i(source[idx..source.len])
164 let pair = this.decode_l(source[idx..source.len])
170 let pair = this.decode_d(source[idx..source.len])
176 let pair = this.decode_s(source[idx..source.len])
```
should be replaced with `source[idx..<source.len]`

**Comment by xmonader** [comment]
Thanks for reporting this :)

**Comment by xmonader** [comment]
Fixed here 595e84d
